model/project_parser.go

```diff
@@ -1370,6 +1373,10 @@ func evaluateBVTasks(tse *taskSelectorEvaluator, tgse *tagSelectorEvaluator, vse
 			}
 		}
 	}
+	// No tasks found should result in an error if there are any tasks defined in the build variant.
+	if len(allNames) == 0 && len(pbv.Tasks) > 0 {
+		evalErrs = append(evalErrs, errors.Errorf("no tasks found for build variant '%s'", pbv.Name))
```

Comment:
I think we should say something like 
`task selectors for build variant '%s'", pbv.Name did not match any tasks`
So that it's differentiated from the 
`buildvariant '%s' must have at least one task` error when no tasks are listed on the build variant at all.